### DISCUSSION TITLE
How to show number of successful tests?

### DISCUSSION THREAD

**Discussion by FieryRMS** [pytest-dev/pytest#10485] [Q&A] [answered]
I am new to pytest so I might be missing something. I know it's possible since its shown in the docs website. But for some reason, it doesn't show for me. My file structure and file content is as follows,
```
src/
├─ tests/
│  ├─ test_1.py
│  ├─ test_2.py
│  ├─ __init__.py
``` 
```python
# test_1.py
assert 1==1
```
```python
# test_2.py
assert 1==2
```
```python
# __init__.py
```

And when I run ```pytest``` from the ```src``` folder, I get the following output,
```
======================================================================== test session starts =========================================================================
platform win32 -- Python 3.10.8, pytest-7.2.0, pluggy-1.0.0
rootdir: C:\Users\FeiryRMS\Projects\src
collected 0 items / 1 error

=============================================================================== ERRORS ===============================================================================
__________________________________________________________________ ERROR collecting tests/test_2.py __________________________________________________________________
tests\test_2.py:1: in <module>
    assert 1==2
E   assert 1 == 2
====================================================================== short test summary info =======================================================================
ERROR tests/test_2.py - assert 1 == 2
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! 
========================================================================== 1 error in 0.32s ========================================================================== 

```

What I was expecting was at least an acknowledgement that there was one successful test. Something like, 
```
======================= 1 failed, 1 passed in 0.12s ========================
or
1 failed, 1 passed in 0.12s
```

Preferably I'd want it to show,
```
SUCCESS tests/test_1.py
ERROR tests/test_2.py - assert 1 == 2
======================= 1 failed, 1 passed in 0.12s ========================
```

It's also showing that it collected 0 items? 

I have tried examples in [getting-started](https://docs.pytest.org/en/7.2.x/getting-started.html) and [output](https://docs.pytest.org/en/7.1.x/how-to/output.html) pages, but nothing seems to work.

**Comment by The-Compiler** [comment, marked as answer]
Like those examples you link, your tests should be in a `def test_...():` function, not at the module body.

**Comment by jwodder** [comment]
Tests aren't supposed to be module-level assertions; they're supposed to be functions containing assertions (or classes with methods containing assertions, but let's not get into that right now).  You need to rewrite `test_1.py` to something like:

```python
def test_good():
    assert 1 == 1
```

and `test_2.py` to:

```python
def test_bad():
    assert 1 == 2
```

Now pytest will be able to collect your tests and report on how many passed & how many failed.
